include/seqan3/search/search.hpp

```diff
@@ -73,6 +73,20 @@ struct search_configuration_validator
         }
     }
 
+    template <typename configuration_t>
+    static auto add_defaults(configuration_t const & cfg)
```

Comment:
Ok, two things:

First this would not be the place to add this. The class is intended for validation and not handling defaults.
Second, you replace are recursive call with another recursive call? Why the change at all then?

If you want to get rid of the recursive call you would rather have two separate functions that indicate with their name what they do and can be called in sequence. 
I propose to directly add a configuration class that offers for now only the two functions. This would be the first part of having later a separate configuration step and a invoking/result_range step.

Reply:
> Second, you replace are recursive call with another recursive call? Why the change at all then?

I meant recursive calls of `search` in favour of delegating this task to a class.

It is true that this class is checking the configuration right now given it's name but I thought this still fits because it is all part of handling the configuration.

> If you want to get rid of the recursive call you would rather have two separate functions that indicate with their name what they do and can be called in sequence.

I also had this in mind but I don't if it's preferable over this solution?

> I propose to directly add a configuration class that offers for now only the two functions. This would be the first part of having later a separate configuration step and a invoking/result_range step.

I agree about separating those two, but why have a separate validation-class and (default-)configuration-class? Why not have one class that handles everything related to the configuration, defaults+validation+whatever? Or did I understand you wrong?

Reply:
Because it becomes quite quickly very complex and it helps for the reader if these kind of things are separated. Moreover, you have more control and can better document the intention of what it does.

Reply:
> I also had this in mind but I don't if it's preferable over this solution?

Yes it is. I also didn't do it in the alignment but would like to change that at some point as it is simply easier to understand and can be better documented.

Reply:
> Yes it is. I also didn't do it in the alignment but would like to change that at some point as it is simply easier to understand and can be better documented.

See the update in commit 18b87913165776e5622fc07e3395752570b033e8
I don't think this makes it easier as it is way less compact. But I also don't care much.